### DISCUSSION TITLE
Fix "transparency issue" with image

### DISCUSSION THREAD

**Discussion by Ry3yr** [ImageMagick/ImageMagick#4700] [Digital Image Processing]
Quick question:

Transparency issues like seen in this image (example, not by me): [Background shines through, cause coloring is inconsistent)
https://web.archive.org/web/20220112082152im_/https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/16e22e67-d6da-4e94-9dd7-52c55c392ab0/dexyqst-7686e101-daaf-4600-9fb1-1f10c742c2e6.png?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOjdlMGQxODg5ODIyNjQzNzNhNWYwZDQxNWVhMGQyNmUwIiwiaXNzIjoidXJuOmFwcDo3ZTBkMTg4OTgyMjY0MzczYTVmMGQ0MTVlYTBkMjZlMCIsIm9iaiI6W1t7InBhdGgiOiJcL2ZcLzE2ZTIyZTY3LWQ2ZGEtNGU5NC05ZGQ3LTUyYzU1YzM5MmFiMFwvZGV4eXFzdC03Njg2ZTEwMS1kYWFmLTQ2MDAtOWZiMS0xZjEwYzc0MmMyZTYucG5nIn1dXSwiYXVkIjpbInVybjpzZXJ2aWNlOmZpbGUuZG93bmxvYWQiXX0.yU9w7KWe6qedp0yR5MySdYQ5MznaVGEOx0LUSIgv4mk

Can they be fixed with imagemagick ?

PS: Whats up with the legacy.IM forum ?
The last post is from Nov last year, and I can't submit a new topic ? Was it laid to rest ?

**Comment by snibgo** [comment]
I don't understand the problem.

The image you link to has no transparency.

**Reply by Ry3yr** [reply]
I'm talking about these white shine through spots near countour edges.
The image here has been flattened, implementing the white bg.
(Hence no transparency)

I made one yesterday that also has that issue (look especially at the left side of her bang-strang, the bg shines through)

https://ry3yr.github.io/OSTR/release/0mine/documents/art/Transparenterror.png

I'd like to fill these spots with neigboring colors to remove those unsightly spots without having to "hunt" for em manually.

**Comment by snibgo** [comment]
```
magick Transparenterror.png -alpha extract a.png
```
a.png is black where you had full transparency, white where you had full opacity, and gray in between. Many pixels within the figure are gray. You could use Gimp or similar to paint those pixels white, then copy the opacity back to the colour image.

Or you could threshold the opacity at 50%, so you no longer have any partial transparency.

Or an algorithm might be devised to automatically remove transparency from pixels that are within the figure.

**Reply by Ry3yr** [reply]
Interesting.
That helps with border contour issues.

Is there a solution for when pixels surrounding the spot are not black, white or grey but any other color ?

Like: detect the neares pixel color near such micro hole and fill it with a neigboring color ?

**Comment by snibgo** [comment]
Did you try the command I showed you? Did you look at the result, a.png? a.png has no colours. Just white, black and shades of gray.

**Reply by Ry3yr** [reply]
Yes, I know. It works, as you said :)
I like your option.

And was wondering if there was a way to skip the "gimp" part.

Ideally I'd look for a command that packs all into one.

(I mean.. I could use

convert -composite -gravity center a.png Transparencyerror.png anew.png

to paste it over... but is that a good option ?)
